### ISSUE TITLE
Memory issues.

### ISSUE THREAD

**Issue by Linden10** [arall/wp-bruteforcer#5]
Happens when using WP-Bruteforcer, brute with a large file (100 mb or so)
and running for sometime.

I've been monitoring the php process and noticing that the memory constantly
increases as it goes on, reaching 1GB after 30 mins or so.

Seems that with each request, memory is not being erased properly (no garbage collection) so
it'll continue to grow until it finishes.

I believe some memory management is needed after a request is made
so that memory isn't hogged/errors out with fatal out of memory errors.

This can be useful for making the script efficient/take up a small footnote.

**Comment by Linden10** [comment]
Hmm...
![Memory](http://i.imgur.com/s9TSkUZ.png)

From running the script for say 30 mins, it's upto 1GB...
But I've been observing the script and seeing it go beyond
that as time goes on. 2, 3, 5 GB and so on when using a large wordlist.

![PeridotReading](http://i.imgur.com/z8rcfoH.png)

I did some reading of your script and from what learned of php...
I may of found the crux of the issue that's causing the memory leak.

There appears to be nothing that's "unset"ing the request per tries in the php script.

For every isset, nothing unsets it so I believe it's stored in memory.
That's a big problem when it comes to large wordlists as the script
goes on for a while.

I am unsure of this (still kinda new to PHP) but
from what I read, this is likely the cause of this.

As such, it's constantly growing in memory with no end.
Hopefully this can help shed some light on where the issue
originates from.

**Comment by arall** [comment]
Thanks for reporting the bug. Yeah, this tool is missing garbage collection. Probably, with some properly placed `unset` methods should be fixed. 
The weird thing is that I did not have those issues when I tested it with large wordlists :/

I will take a look whenever I have some time. Thanks again.

**Comment by Linden10** [comment]
Are you perhaps on Windows or Linux/Unix?

It could be something on that not happening with
that or...some sort of setting with the PHP build you have.

Honestly I don't know, odd it's not happening to you.
Still thanks and hopefully you can figure a way to resolve this in time!

**Comment by arall** [comment]
Linux/Unix, always :P

**Comment by Linden10** [comment]
Makes sense, I'm using Windows 7 and the latest version of PHP 7 [PHP 7.0.9] 64-bit.
(I have both PHP 5 (kinda old) and PHP 7 (recently upgraded to this), just so you know...the
issue affects both, with PHP 5 having fatal error out of memory issues [stopping the script] while PHP 7 having none which is awesome. Still the memory leak exists).

Seeing you don't have Windows, I can only think of using a virtual machine
to find the flaws...though if you do have a computer around with it, that'll do too.

Even if you don't need it, hopefully you can figure something out!
Thanks for helping btw!

**Comment by Linden10** [comment]
Hmm...small update:

I tested this script on a linux disto (debian) with php5
and the memory leak also happens on that as well. =/

Sometime later after the script's memory leak kept on going, linux killed
the php process... (I may of need to change the mem limit in php's config
or something but still...) so I have no idea what disto/php settings you have
that doesn't cause the memory leak issues for you (unless you're not letting
the process run for a long while, like 30 mins to an hour or so...IDK why you're
not experiencing the problem.) but this issue is still present in linux.

If you're using the default settings for WP-Bruteforcer, maybe you won't notice it...
But try changing the number of tries like say to 500 or 1000 (Up to 1000 is okay).

Maybe you're not seeing it increase as much with the default 100 tries...
Try a large wordlist (50 to 100mb or so) with it...you'll most likely see it...likely.
